### ISSUE TITLE
Rename discovered thing

### ISSUE THREAD

**Issue by jarrodlucia** [mozilla-iot/gateway#874]
use-case: I have discovered light bulbs that I move around the house. Changing the name involves currently deleting and re-adding to change the name in the web front end.

Have an update to name so movement of devices and name changes is easier.

**Comment by benfrancis** [comment]
Thanks for reporting this, have been meaning to implement this for a while.

**Comment by benfrancis** [comment]
OK, now we have the overflow menu in the thing detail view, I think we can add an "edit thing" menu entry which allows you to edit the name of a thing. Once #491 is implemented this view can also be used to edit the icon of a thing by choosing a capability type or uploading a custom icon.

Note that this re-uses the UI element from the add thing dialog.

![rename_thing](https://user-images.githubusercontent.com/552417/40560915-534934d2-6053-11e8-9b64-88e3daf67993.png)

I also suggest moving the "remove thing" feature to the overflow menu as well, instead of a long-press/right-click on the main things screen. This could be done as part of this issue, or as a separate issue, but it will free up that interaction for something else (e.g. #1015).

**Comment by mrstegeman** [comment]
@benfrancis How would you rename an on/off switch or something else without a detail view?

**Comment by benfrancis** [comment]
UX spec [added](https://docs.google.com/presentation/d/1rMeg6hep8JQflqwbiNIbhSoK9StCeVZ2npphmQHKda4/edit#slide=id.g3b5aa6b68a_0_10).

**Comment by benfrancis** [comment]
> @benfrancis How would you rename an on/off switch or something else without a detail view?

Good point. Maybe we need to add the detail view to all thing types now. We're probably going to need to do that when we implement the capabilities system anyway. (I think you may have recently asked sogaani not to do this, so sorry about that!)
